src/ui/render.ts

```diff
@@ -396,34 +419,77 @@ export default class UI {
 
   private onCoreSetCell(event: LightsheetEvent) {
     const payload = event.payload as CoreSetCellPayload;
-    // Get HTML elements and (new) IDs for the payload's cell and row.
-    const elInfo = LightSheetHelper.getElementInfoForSetCell(payload);
-
-    if (!elInfo.rowDom) {
-      const row = this.addRow(payload.indexPosition.row);
-      elInfo.rowDom = row;
-      row.id = elInfo.rowDomId;
+    // Create new columns if the column index is greater than the current column count.
+    const newColumns = payload.indexPosition.column - this.getColumnCount() + 1;
+    for (let i = 0; i < newColumns; i++) {
+      this.addColumn();
     }
-    if (!elInfo.cellDom) {
-      elInfo.cellDom = this.addCell(
-        elInfo.rowDom!,
-        payload.indexPosition.column,
-        payload.indexPosition.row,
-        payload.formattedValue,
-        payload.keyPosition.columnKey?.toString(),
-      );
+
+    const newRows = payload.indexPosition.row - this.getRowCount() + 1;
+    for (let i = 0; i < newRows; i++) {
+      this.addRow();
     }
 
-    elInfo.cellDom.id = elInfo.cellDomId;
-    elInfo.rowDom.id = elInfo.rowDomId;
+    // Get HTML elements and (new) IDs for the payload's cell and row.
+    const elInfo = this.getElementInfoForSetCell(payload);
+
+    elInfo.cellDom!.id = elInfo.cellDomId;
+    elInfo.rowDom!.id = elInfo.rowDomId;
 
     // Update input element with values from the core.
-    const inputEl = elInfo.cellDom.firstChild! as HTMLInputElement;
+    const inputEl = elInfo.cellDom!.firstChild! as HTMLInputElement;
     inputEl.setAttribute("rawValue", payload.rawValue);
     inputEl.setAttribute("resolvedValue", payload.formattedValue);
     inputEl.value = payload.formattedValue;
   }
 
+  private getElementInfoForSetCell = (payload: CoreSetCellPayload) => {
```

Comment:
Actually this whole method is wrong, Ive "NUKED" it in my setStyle issue, but that is still draft. 
But nice improvements.

Reply:
I will keep it for now